### PR TITLE
Review component package grammar

### PR DESCRIPTION
**Pull request by bryanuribe**
**Description:**
This PR improves the documentation for the component package by fixing grammar mistakes. Changes were made to several modules in the component package, focusing on improving the readability of the comments. There was a high emphasis on maintaining the semantic meaning intended by the author of the comments.  

**Link to tracking Issue:**
Addresses issue [#3049](https://github.com/open-telemetry/opentelemetry-collector/issues/3049).

### CODE REVIEW

File: component/component.go [hunk 1]
```diff
@@ -33,27 +33,27 @@ import (
...
+	// accepted data in any way, it should not accept it anymore.
 	//
-	// If there are any background operations running by the component they must be aborted as soon as possible.
-	// Remember that if you started any long-running background operation from the Start() method that operation
-	// must be also cancelled. If there are any buffer in the component, it should be cleared and the data sent
-	// immediately to the next component.
-	//
-	// Once the Shutdown() method returns the component's lifecycle is completed. No other
+	// If there are any background operations running by the component they must be aborted as 
+	// soon as possible. Remember that if you started any long-running background operations from
```
Comment (bogdandrutu) [inline component/component.go:52]: s/soon as possible./before this function returns.?

**Review by bogdandrutu — APPROVED** [review]
Thanks.

**Review by carlosalberto — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by bogdandrutu** [comment]
Please fix lint issues:
```
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector/opentelemetry-collector'
component/component.go:44: File is not `gofmt`-ed with `-s` (gofmt)
	// to Start() function since that context will be cancelled soon and can abort the long-running 
make[2]: *** [Makefile.Common:26: lint] Error 1
make[1]: *** [Makefile:179: for-all] Error 2
make: *** [Makefile:99: golint] Error 2
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector/opentelemetry-collector'
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector/opentelemetry-collector'
Error: Process completed with exit code 2.
```

File: component/component.go (hunk 1 above)
Comment (bryanuribe) [reply component/component.go]: Fixed this.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**